src/libraries/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.UnitTests/System/Runtime/InteropServices/NFloatTests.GenericMath.cs

```diff
@@ -217,6 +217,22 @@ public static void AdditiveIdentityTest()
         // IBinaryNumber
         //
 
+        [Fact]
+        public static void AllBitsSetTest()
+        {
+            if (NFloat.Size == sizeof(float))
+            {
+                Assert.Equal(0xFFFFFFFF, BitConverter.SingleToUInt32Bits((float)BinaryNumberHelper<NFloat>.AllBitsSet));
+                Assert.Equal((uint)0, (uint)(BitConverter.SingleToUInt32Bits((float)BinaryNumberHelper<NFloat>.AllBitsSet) + 1));
+            }
+            else
+            {
+                Assert.Equal(0xFFFFFFFFFFFFFFFF, BitConverter.DoubleToUInt64Bits((double)BinaryNumberHelper<NFloat>.AllBitsSet));
+                Assert.Equal((ulong)0, (ulong)(BitConverter.DoubleToUInt64Bits((double)BinaryNumberHelper<NFloat>.AllBitsSet) + 1));
+            }
+
```

Comment:
nit: For consistency with the other tests, do:
```suggestion
            if (Environment.Is64BitProcess)
            {
                Assert.Equal(0xFFFFFFFFFFFFFFFF, BitConverter.DoubleToUInt64Bits((double)BinaryNumberHelper<NFloat>.AllBitsSet));
                Assert.Equal((ulong)0, (ulong)(BitConverter.DoubleToUInt64Bits((double)BinaryNumberHelper<NFloat>.AllBitsSet) + 1));
            }
            else
            {
                Assert.Equal(0xFFFFFFFF, BitConverter.SingleToUInt32Bits((float)BinaryNumberHelper<NFloat>.AllBitsSet));
                Assert.Equal((uint)0, (uint)(BitConverter.SingleToUInt32Bits((float)BinaryNumberHelper<NFloat>.AllBitsSet) + 1));
            }
```

Reply:
I'm also not convinced the assert that `AllBitsSet + 1 == 0` is adding much value.

It's not harming anything either however. You could simplify it a bit by using `0UL`, `0U`, `1UL`, and `1U` however.

Reply:
Thanks for the suggestions, I'll add them all. 

Re: the tests, I wanted to include a test case that wasn't a copy paste of my implementation logic. For example, if I implement `char.AllBitsSet` as `0xFF`, then make a test case that only does `Assert.Equal(0xFF, char.AllBitsSet)`, the test case passes, and the bug goes unnoticed. With `AllBitsSet + 1 == 0`, I can confirm that my implementation logic is actually correct. This example was actually a real bug I ran into, as I was going off of my memory of C and assuming `char` was one byte.

Reply:
Makes sense, but I'd note that testing `Zero == ~AllBitsSet` might be a better one.

There are theoretically types where `AllBitsSet + 1` won't become the same as "zero", for many possible reasons. However, for bitwise manipulation the implication is that `Zero` is "no bits set" and `~` being a one's complement should therefore give you "all bits set" as the result (regardless of underlying format representation), regardless of there being other unrelated bits (like flags, etc).

Reply:
Good point, I'll switch to that method